symengine/basic-inl.h

```diff
@@ -79,7 +79,13 @@ inline void hash_combine_impl(hash_t &seed, const std::string &s)
 
 inline void hash_combine_impl(hash_t &seed, const double &s)
 {
-    hash_combine(seed, static_cast<hash_t>(s));
+    union {
+        hash_t h;
+        double d;
+    } u;
+    u.h = 0u;
+    u.d = s;
```

Comment:
Why is the `union` required if we aren't using `u.d` anywhere. Also, won't this overwrite `u.h`.